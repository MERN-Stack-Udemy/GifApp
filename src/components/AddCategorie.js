import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategorie = ( {setCategories} ) => {

  const [inputvalue, setInputvalue] = useState(''); // ''

  const handleChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputvalue.trim().length > 2){
      setCategories(categories => [...categories, inputvalue])
      setInputvalue( "" );
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type = "text" 
        value = {inputvalue} 
        onChange = {handleChange} />
    </form>
  );

};

AddCategorie.propTypes = {
  setCategories : PropTypes.func.isRequired,
}
