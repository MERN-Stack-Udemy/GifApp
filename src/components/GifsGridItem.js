import React from 'react'

export const GifsGridItem = ( {url, title} ) => {
  
  return (
    <div className="card animate__bounceIn">
      <img 
        src={url}
        alt= {title}
      />
      <p>{title}</p>
    </div>
  )
}
