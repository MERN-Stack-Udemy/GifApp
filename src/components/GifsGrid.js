import React from 'react'

import { useFetchGifs } from '../Hooks/useFetchGifs'
import { GifsGridItem } from './GifsGridItem';

export const GifsGrid = ({ category }) => {

  const {data, loading} = useFetchGifs(category);
  console.log({data, loading})
  return (
    <>
      <h3  className= "animate__jello">{category}</h3>
      { loading && <p>Loading....</p>}
      <div className="card-grid">
        {data.map( (img) => (
          <GifsGridItem 
            key = {img.id} 
            {...img}
          />
        ))}
        
      </div>
    </>
  )
}
