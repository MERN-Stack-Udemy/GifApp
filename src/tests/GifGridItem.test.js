import { shallow } from "enzyme"
import { GifsGridItem } from "../components/GifsGridItem"


describe('Test about GifsGridItem', () => {
  
  test('test snapshot of GifsGritItem', () => {
    const image= { url: 'holamundo'};
    const title = 'test demo'
    const wrapper = shallow(<GifsGridItem image={image} title={title}/>);
    expect(wrapper).toMatchSnapshot();
  })
  

})
