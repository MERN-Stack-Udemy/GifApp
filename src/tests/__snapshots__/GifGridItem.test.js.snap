// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test about GifsGridItem test snapshot of GifsGritItem 1`] = `
<div
  className="card animate__bounceIn"
>
  <img
    alt="test demo"
    src="holamundo"
  />
  <p>
    test demo
  </p>
</div>
`;
